jsparagus/parse_table.py

```diff
@@ -20,6 +20,9 @@
 # StateAndTransitions objects are indexed using a StateId which is an integer.
 StateId = int
 
+# Action or ordered sequence of action which have to be performed.
+DelayedAction = typing.Union[Action, typing.Tuple[Action, ...]]
```

Comment:
`actions.Seq` has restriction into what can be executed while the tuple is a sequence of actions with no restriction, apart from existing.

Reply:
Hmm. OK. It would be better to generalize `Seq` to handle these additional cases. Most of the methods can just delegate to `self.actions` in obvious ways, like `return all(a.is_condition() for a in self.actions)`. And I think it would be OK for some methods to throw NotImplementedError when the sequence is too complicated for the method to give a single complete answer.

However, we probably shouldn't be working on that kind of detail right now, so this is very optional.